code/modules/urist/antagonist/revenant/archetypes/chimera.dm

```diff
@@ -101,51 +101,103 @@
 /mob/proc/bsrevenant_minor_heal()
 	set name = "Fleshmend"
 	set category = "Anomalous Powers"
-	set desc = "Heal minor damage, stop bleeding. Use multiple times per each wound."
-
-	BSR_ABORT_IF_DEAD_PRESET(src)
+	set desc = "Unnatural regeneration. Revive on death, heal wounds/bloodloss otherwise."
 
 	// if TRUE, we're still waiting for a handler
 	var/pending = TRUE
 
+	var/datum/bluespace_revenant/revenant = src?.mind?.bluespace_revenant
+	var/valid_revenant = istype(revenant)
+
 	var/mob/living/carbon/human/H = src
 
 	if(istype(H))
+		// First things first - if they are dead, revive
 		if(pending)
-			for(var/obj/item/organ/I in H.internal_organs)
-				if(I.damage >= 5)
-					I.damage = max(I.damage - 5, 0)
-					to_chat(H, SPAN_NOTICE("Your [I.name] itches as you knit its wounds shut."))
-					pending = FALSE
-					break
+			if(H.stat == DEAD)
+				H.adjustOxyLoss(-rand(15,20))
+				H.basic_revival()
+				H.visible_message(SPAN_NOTICE("\The [H] shudders violently!"))
+
+				to_chat(H, SPAN_NOTICE("You have revived yourself."))
+				to_chat(H, SPAN_WARNING("By reviving, you have significantly violated the laws of reality!"))
+
+				// Short-circuit to a return; because reviving is more distortey
+				if(valid_revenant)
+					revenant.total_distortion += BSR_DISTORTION_GROWTH_OVER_SECONDS(1500, BSR_DEFAULT_DISTORTION_PER_TICK, BSR_DEFAULT_DECISECONDS_PER_TICK)
+					return TRUE
 
 		for(var/obj/item/organ/external/E in H.bad_external_organs)
 			if(!pending)
 				break
 
-			if((E.status & ORGAN_ARTERY_CUT))
-				to_chat(H, SPAN_NOTICE("You mend the torn artery in your [E.name], stemming the worst of the bleeding."))
-				E.status &= ~ORGAN_ARTERY_CUT
-				pending = FALSE
-				break
+			if(prob(70))
+				continue
 
-			for(var/datum/wound/W in E.wounds)
-				if(W.bleeding())
-					W.bleed_timer = 0
-					E.status &= ~ORGAN_BLEEDING
-					to_chat(H, SPAN_NOTICE("You knit together severed veins, stemming the bleeding from your [E.name]."))
-					pending = FALSE
-					break
+			E.rejuvenate()
```

Comment:
This is going to also full heal all organs inside each limb. bad_external_organs also includes any limb with internal damage (excluding brain damage) even if it lacks external damage.

Reply:
That's kind of the point; I would do that by individual organs as prior except, err, that code didn't work properly fsr and would routinely block all other healing (which is why there's a prob onnit; stochastically it will eventually heal everything, just potentially on multiple tries)

Reply:
Since neither this or the blood restoration trigger any distortion increases burn damage is healed for free if you just cast this enough.

Reply:
Never mind I missed the 300 seconds catch all increase. The cost being detached from the amount of damage healed should still be considered though.

Reply:
Nope, they do - that's what the fallthrough case at the very bottom does. Unless a heal-type has a specific custom cost, it gives it a generic cost.